data_model/src/block.rs

```diff
@@ -0,0 +1,662 @@
+    /// Array of all rejected transactions in this block.
+    pub rejected_transactions: Vec<VersionedRejectedTransaction>,
+    /// Array of all valid transactions in this block.
+    pub transactions: Vec<VersionedValidTransaction>,
+    /// Signatures of peers which approved this block.
+    pub signatures: SignaturesOf<Self>,
+    /// Event recommendations.
+    pub event_recommendations: Vec<Event>,
+}
+
+/// Revalidate the block that was sent through the network by transforming it back into [`ValidBlock`]
+#[version_with_scale(n = 1, versioned = "VersionedCandidateBlock")]
+#[derive(Debug, Clone, Decode, Encode, IntoSchema)]
+pub struct CandidateBlock {
+    /// Block header
+    pub header: BlockHeader,
+    /// Array of rejected transactions.
+    pub rejected_transactions: Vec<VersionedSignedTransaction>,
+    /// Array of all transactions in this block.
+    pub transactions: Vec<VersionedSignedTransaction>,
+    /// Signatures of peers which approved this block.
+    pub signatures: SignaturesOf<Self>,
+    /// Event recommendations.
+    pub event_recommendations: Vec<Event>,
+}
+
+/// The `CommittedBlock` struct represents a block accepted by consensus
+#[version_with_scale(n = 1, versioned = "VersionedCommittedBlock")]
+#[derive(
+    Debug, Clone, Decode, Encode, IntoSchema, Serialize, Deserialize, PartialEq, Eq, Display, Hash,
+)]
+#[display(fmt = "({header})")]
+pub struct CommittedBlock {
+    /// Block header
+    pub header: BlockHeader,
+    /// Array of rejected transactions.
+    pub rejected_transactions: Vec<VersionedRejectedTransaction>,
+    /// array of transactions, which successfully passed validation and consensus step.
+    pub transactions: Vec<VersionedValidTransaction>,
+    /// Event recommendations.
+    pub event_recommendations: Vec<Event>,
+    /// Signatures of peers which approved this block
+    pub signatures: SignaturesOf<Self>,
+}
+
+/// Block state used to transfer accepted by consensus block through network to the other peers.
+/// This block state is not entirely trusted and require hash revalidation to obtain `CommittedBlock`.
+#[version_with_scale(n = 1, versioned = "VersionedCandidateCommittedBlock")]
+#[derive(Debug, Clone, Decode, Encode, IntoSchema, Serialize)]
+pub struct CandidateCommittedBlock {
+    /// Block header
+    pub header: BlockHeader,
+    /// Array of rejected transactions.
+    pub rejected_transactions: Vec<VersionedRejectedTransaction>,
+    /// Array of transactions, which successfully passed validation and consensus step.
+    pub transactions: Vec<VersionedValidTransaction>,
+    /// Event recommendations.
+    pub event_recommendations: Vec<Event>,
+    /// Signatures of peers which approved this block
+    pub signatures: SignaturesOf<Self>,
+}
+
+impl BlockHeader {
+    /// Checks if it's a header of a genesis block.
+    #[inline]
+    pub const fn is_genesis(&self) -> bool {
+        self.height == 1
+    }
+}
+
+impl PartialOrd for BlockHeader {
```

Comment:
I have done such reorganization, however i preferred to leave block names as is, because this way it is more consistent with change your proposed for `schema.json` about having unambiguous names.